### PR TITLE
Add CreateOutboundEmailAsync

### PR DESCRIPTION
**Pull request by RobertSmits**

### CODE REVIEW

File: FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            EmailConfigId = emailConfigId;
+            FirstResponseDueBy = firstResponseDueBy;
+            GroupId = groupId;
+            Tags = tags;
+            Subject = subject;
+            TicketType = ticketType;
+            Files = files;
+        }
+
+        /// Name of the requester
```
Comment (trejjam) [inline FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs:39]: Please use valid [documentation-comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments)

```suggestion
        /// <summary>
        /// Name of the requester
        /// </summary>
```

**Comment by trejjam** [comment]
Hi, thanks for the PR. It looks good to me. Please, just improve the comments in the request.

File: FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        [JsonProperty("name")]
+        public string? RequesterName { get; }
+
+        /// Email address of the requester. If no contact exists with this email address in Freshdesk, it will be added as a new contact.
+        [JsonProperty("email")]
+        public string? Email { get; }
+
+        /// Subject of the ticket. The default Value is null.
+        [JsonProperty("subject")]
+        public string? Subject { get; }
```
Comment (trejjam) [inline FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs:49]: Can you check the type in the constructor with the property type?

File: FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs (hunk 1 above)
Comment (RobertSmits) [reply FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs:40]: The CreateOutboundEmailRequest was copied from CreateTicketRequest, which als doesn't have xml comments.
I have fixed them in CreateOutboundEmailRequest.

File: FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs (hunk 2 above)
Comment (RobertSmits) [reply FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs:55]: I have kept the `string?` type as in CreateTicketRequest, and the API docs say default value is null.
However api docs also sugest this field is mandatory, I'll check tomorrow what would happen if I leave this fied empty.
Dependig on the result I will change it to `string` or keep it `string?`.

Comment (trejjam) [reply FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs:55]: That will be awesome. Thanks

File: FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+        /// Email address of the requester. If no contact exists with this email address in Freshdesk, it will be added as a new contact.
+        /// </summary>
+        [JsonProperty("email")]
+        public string? Email { get; }
+
+        /// <summary>
+        /// Subject of the ticket. The default Value is <see langword="null"/>.
+        /// </summary>
+        [JsonProperty("subject")]
+        public string Subject { get; }
```
Comment (trejjam) [inline FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs:55]: Are you sure that it is a not null property when docs say the default value is `null`?

Comment (RobertSmits) [reply FreshdeskApi.Client/Tickets/Requests/CreateOutboundEmailRequest.cs:55]: image
The Freshdesk API documentation itself is a little conflicted about this. It says the subject is mandatory, however the default value is also null. Calling the API endpoint without a subject also doesn't work, which also suggests it shouldn't be left empty.

**Comment by trejjam** [comment]
Thank you, merged :tada:

**Pull request merged**